src/SDDPoptimize.jl

```diff
@@ -17,39 +163,71 @@ scenario according to the current value functions.
 Parameters:
 - model (SPmodel)
     the stochastic problem we want to optimize
-    
+
 - param (SDDPparameters)
     the parameters of the SDDP algorithm
 
+- n_iterations (Int) - Default is 20
+    Maximum number of iterations to run
+
+- display (Bool) - Default is false
+    If specified, display progression in terminal
+
 
 Returns :
-- V::Array{PolyhedralFunction}
+- V (Array{PolyhedralFunction})
     the collection of approximation of the bellman functions
 
+- problems (Array{JuMP.Model})
+    the collection of linear problems used to approximate
+    each value function
+
 """
-function optimize(model::SPmodel,
-                  param::SDDPparameters)
-    # TODO initialization (V and so on)
-    
-    stopping_test::Bool = false;
-    iteration_count::int = 0;
-    
+function optimize(model::SDDP.SPModel,
+                  param::SDDP.SDDPparameters,
+                  n_iterations=20,
+                  display=true)
+
+    # Initialize value functions:
+    V, problems = initialize_value_functions(model, param)
+
+    if display
+      println("Initialize cuts")
+    end
+
+    # Build given number of scenarios according to distribution
+    # law specified in model.noises:
+    aleas = simulate_scenarios(model.noises ,
+                                (model.stageNumber,
+                                 param.forwardPassNumber,
+                                 model.dimNoises))
+
+    stopping_test::Bool = false
+    iteration_count::Int64 = 0
+
     n = param.forwardPassNumber
-    
-    while(!stopping_test)
```

Comment:
I think it could be great to let a general stopping test even if we only use the number of iteration for the moment. We can define stopping test with 2 parameters. The first one is the type of stopping test and the second one, a characteristic.  For example, (1,20) means iteration cout -- 20;     (1,100) means iteration count --100;      (2,0.001) means gap between bounds is 0.001.